api/tools/src/main/java/org/pmiops/workbench/tools/CommandLineToolConfig.java

```diff
@@ -0,0 +1,76 @@
+package org.pmiops.workbench.tools;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.apache.ApacheHttpTransport;
+import com.google.gson.Gson;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.ConfigDao;
+import org.pmiops.workbench.db.model.Config;
+import org.springframework.context.annotation.*;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import static org.springframework.context.annotation.FilterType.ASSIGNABLE_TYPE;
+
+/**
+ * Contains Spring beans for dependencies which are different for classes run in the context of a
+ * command-line tool versus a WebMVC request handler.
+ * <p>
+ * The main difference is that request-scoped dependencies cannot be used from a command-line
+ * context.
+ */
+@Configuration
+@EnableJpaRepositories({"org.pmiops.workbench.db.dao"})
+public class CommandLineToolConfig {
+
+  /**
+   * Loads the GSuite admin service account key from local disk.
+   *
+   * This needs to be annotated with @Lazy so only classes that use it (e.g. BackfillGSuiteUserData
+   * which requires a WorkbenchConfig instance) will trigger the file load attempt.
+   *
+   * Any command-line tool which loads this bean needs to be called from a project.rb command
+   * which is preceded with "get_gsuite_admin_key" to ensure the local key file si populated.
+   *
+   * @return
+   */
+  @Lazy
+  @Bean
+  GoogleCredential googleCredential() {
+    try {
+      String saKeyPath = "src/main/webapp/WEB-INF/gsuite-admin-sa.json";
```

Comment:
hmm... I think this would really only work for the test environment. This file only exists after you've run dev-up (or possibly if you ran a build on a given environment, which would be unusual). We'd either need to `gsutil cp` into here before running the tool - or probably better somehow load the relevant Config for the appropriate environment, and grab the `gs://` path from there. Pulling from gs:// in here would be preferred, as we'd rather not pull the prod gsuite admin key file (for example) onto a dev workstation.

Alternatively, you could instead try to pass this gs:// path as a flag to your CLI, then somehow get it into here, but this might just be creating more work for the next config value someone adds (or CLI someone writes).

Reply:
AFAICT this would technically have worked correctly as written – the backfill_gsuite_user_data method in devstart.rb contained a "get_gsuite_admin_key" command, which fetches and stores the gsuite key for the current project. Yes, it would have stored the prod key locally when run against the prod project... but it doesn't seem to be any different than how we store the prod key locally when deploying the App Engine project, https://github.com/all-of-us/workbench/blob/c2837ea680eac13cf03b32feadacbf5f8b8fae31/api/libproject/devstart.rb#L1382 but maybe I'm misunderstanding what 'deploy_app' is used for.

Anyway – I was able to change the command-line config to instead fetch the GSuite admin credentials from GCS, which feels like a much better solution in the long run.

Reply:
Ah I missed that line (looks like it was commented out when I was reviewing). Yeah this would have worked.

Note: Minor distinction - when deploy_app runs locally, it typically gets run inside of a "hermetic" docker container (via this deploy stuff: https://github.com/all-of-us/workbench/blob/c2837ea680eac13cf03b32feadacbf5f8b8fae31/deploy/README.md), so the key files aren't getting pulled into your development git client, at least.